.github/workflows/build.yml

```diff
@@ -0,0 +1,134 @@
+name: Build, Test and Release
+
+on: push
+
+permissions: read-all
+
+# update in build.yml and codeql.yml at same time
+env:
+  PROTOC_VERSION: 21.3
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    env:
+      PROTOC: protoc-release/bin/protoc
+      PROTOC_INC: protoc-release/include
+      PROTOC_PLATFORM: linux-x86_64
+
+    steps:
+
+      - uses: actions/checkout@v3
+
+      - uses: actions/setup-node@v3
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}
+          node-version: 16
+          cache: ${{ !env.ACT && 'npm' || '' }} # cache API not available in ACT
+
+      - uses: bazelbuild/setup-bazelisk@v2
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Install Protoc
+        run: |
+          echo "Fetching protoc"
+          curl --header 'Authorization: Bearer ${{ secrets.GITHUB_TOKEN }}' \
+            -L https://github.com/protocolbuffers/protobuf/releases/download/v${PROTOC_VERSION}/protoc-${PROTOC_VERSION}-${PROTOC_PLATFORM}.zip \
+            --output protoc-release.zip
+          unzip protoc-release.zip -d protoc-release
+          rm protoc-release.zip
```

Comment:
There was no need for that whole $GITHUB_WORKSPACE stuff since it's all being executed in that directory as it is.